UI/PlayerListWnd.cpp

```diff
@@ -190,8 +190,14 @@ namespace {
 
             // empire color squares
             for (const auto& empire_id : m_empire_ids) {
+                int eliminated_empires_to_left = 0;
+                for (const auto& empire : Empires()) {
+                    if (empire.second->EmpireID() >= empire_id) break;
+                    if (empire.second->Eliminated()) eliminated_empires_to_left++;
+                }
+
                 const GG::Clr& square_color = GetEmpire(empire_id)->Color();
-                GG::Pt square_ul = ul + GG::Pt(GG::X(empire_id * (ICON_SIZE + PAD)), GG::Y0);
+                GG::Pt square_ul = ul + GG::Pt(GG::X((empire_id - eliminated_empires_to_left) * (ICON_SIZE + PAD)), GG::Y0);
```

Comment:
should probably not assume empire ids are sequential starting from 0 anyway...

Reply:
Ok, I'll switch to a different method.